.github/workflows/docbuild.yaml

```diff
@@ -38,12 +46,29 @@ jobs:
           mkdir -p _build/src
           make html
           touch _build/html/.nojekyll
+      - name: Archive cache
+        if: github.repository == 'project-chip/connectedhomeip' && github.event_name == 'push' && github.ref_name == 'master'
+        uses: actions/upload-artifact@v2
+        with:
+          name: html
+          path: matter/docs/_build/html
```

Comment:
Why do you need to upload it and download instead of deploying it right away if this is push event?

Reply:
This was in case further separation would be required down the line and fully skip the publish job on PRs, but if preferred I can revert this change and leave it for when it is actually needed

Reply:
Not sure how big the docs already are so maybe it doesn't matter, but I always try to reduce the CI time as our CI is already quite overloaded. :)

Reply:
When testing on my fork it used 27 seconds on the archive process and 15 seconds downloading. These timings will increase with the doc size, so I will make some changes. Thanks!

Reply:
Thank you!